### PR TITLE
Improve QA workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
Please see comments below.

### CODE REVIEW

File: .github/workflows/qa.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     strategy:
       matrix:
         php: [ '8.1', '8.2', '8.3' ]
-        lowest: [ '0','1' ]
+        dependencies: [ highest, lowest ]
```
Comment (szepeviktor) [inline .github/workflows/qa.yaml:8]: Display meaningful words in workflow run titles, no more binary code.

File: .github/workflows/qa.yaml [hunk 2]
```diff
@@ -16,21 +16,20 @@ jobs:
           php-version: ${{ matrix.php }}
           extensions: mbstring, intl
           coverage: xdebug
-          tools: composer:v2
```
Comment (szepeviktor) [inline .github/workflows/qa.yaml:19]: Composer v2 is the default years ago.

File: .github/workflows/qa.yaml [hunk 3]
```diff
@@ -16,21 +16,20 @@ jobs:
...
 
-      - name: Download phing
-        run: mkdir bin && curl -L -C - -s -S -o bin/phing https://github.com/phingofficial/phing/releases/download/3.0.0-rc6/phing-3.0.0-RC6.phar && chmod +x bin/phing
-
-      - name: Composer update
-        run: composer update
+      - name: Install phing
         env:
-          COMPOSER_PREFER_LOWEST: ${{ matrix.lowest }}
+          PHING_VERSION: '3.0.0'
```
Comment (szepeviktor) [inline .github/workflows/qa.yaml:25]: Upgrade Phing.

File: .github/workflows/qa.yaml [hunk 4]
```diff
@@ -16,21 +16,20 @@ jobs:
...
+        run: |
+          mkdir -p ~/.local/bin
```
Comment (szepeviktor) [inline .github/workflows/qa.yaml:27]: This is already on `PATH`.
https://github.com/actions/runner-images/pull/2491

File: .github/workflows/qa.yaml [hunk 5]
```diff
@@ -16,21 +16,20 @@ jobs:
...
+          curl -L -C - -s -S -o ~/.local/bin/phing https://github.com/phingofficial/phing/releases/download/${PHING_VERSION}/phing-${PHING_VERSION}.phar
+          chmod +x ~/.local/bin/phing
 
-      - name: Composer install
-        run: bin/phing composer:install
```
Comment (szepeviktor) [inline .github/workflows/qa.yaml:33]: `phing composer:install` did nothing at all

**Comment by jawira** [comment]
Hi, I will accept your PR and make tome changes later.
Thank you for your contribution :)

**Comment by szepeviktor** [comment]
Glad to contribute.

**Comment by jawira** [comment]
@szepeviktor I followed your suggestions and improved pipelines even more. Thanks! :)

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**